runtime/src/main/java/io/quarkus/vault/runtime/VaultKvManager.java

```diff
@@ -60,15 +93,16 @@ private Map<String, String> convert(Map<String, Object> map) {
 
     @Override
     public Uni<Map<String, String>> readSecret(String path) {
-        return readSecretJson(path).map(this::convert);
+        return readValues(mount, path).map(this::convert);
     }
 
     @Override
     public Uni<Map<String, Object>> readSecretJson(String path) {
-        return vaultAuthManager.getClientToken(vaultClient).flatMap(token -> {
-
-            String mount = getConfig().kvSecretEngineMountPath();
+        return readValues(mount, path);
+    }
 
+    private Uni<Map<String, Object>> readValues(String mount, String path) {
```

Comment:
Can't you name this `readSecret` as well? It has a mount path to handle the overload.